### PR TITLE
Newfeature

### PR DESCRIPTION
**Pull request by boxiXia**
Added external force for the mass: 
Vec force_extern
This attribute can be accessed publicly.
This feature is helpful for a simulation that involves applying a persistent external force. for example static/dynamics analysis.

### CODE REVIEW

File: include/Titan/vec.h [hunk 1]
```diff
@@ -148,6 +148,18 @@ class Vec {
         return sqrt(pow(data[0], 2) + pow(data[1], 2) + pow(data[2], 2));
     } // gives vector norm
 
+    CUDA_CALLABLE_MEMBER Vec normalize() {
+        double n = this->norm();
+        //if (n<1e-8)
+        //{// Todo: change this
+        //    n = 1e-8;// add for numerical stability
+        //}
+        data[0] = data[0] / n;
```
Comment (jacobaustin123) [inline include/Titan/vec.h:156]: Change to something like:

```c++
double n = this->norm();
if (n < 1e-8) throw std:: overflow_error("Overflow encountered in vector normalization");
data = data / n;
return *this;
```

**Comment by jacobaustin123** [comment]
@boxiXia happy to merge this if you make the small requested change.

**Comment by boxiXia** [comment]
@jacobaustin123 added the requested change, thanks!

**Comment by jacobaustin123** [comment]
@boxiXia For the normalize function, do you want it to normalize the vector in place, or return a new vector?

**Comment by jacobaustin123** [comment]
@boxiXia I added this feature as part of a fairly large redesign of the library. Let me know if the new method works for you. To adapt existing code, you just need to use the titan namespace (using namespace titan) and check the few functions that have been renamed.

**Comment by boxiXia** [comment]
Thanks! This should work.

**Pull request closed without merging**